addons/sale_project/models/project.py

```diff
@@ -646,51 +648,64 @@ def _get_revenues_items_from_invoices(self, excluded_move_line_ids=None, with_ac
-            currency_ids = OrderedSet(iml['currency_id'] for iml in invoices_move_line_read)
-            move_ids = set()
-            amount_invoiced = amount_to_invoice = 0.0
-            for moves_read in invoices_move_line_read:
-                currency = self.env['res.currency'].browse(moves_read['currency_id']).with_prefetch(currency_ids)
-                price_subtotal = currency._convert(moves_read['price_subtotal'], self.currency_id, self.company_id)
-                # an analytic account can appear several time in an analytic distribution with different repartition percentage
-                analytic_contribution = sum(
-                    percentage for ids, percentage in moves_read['analytic_distribution'].items()
-                    if str(self.analytic_account_id.id) in ids.split(',')
-                ) / 100.
-                move_ids.add(moves_read['move_id'])
-                if moves_read['parent_state'] == 'draft':
-                    if moves_read['move_type'] == 'out_invoice':
-                        amount_to_invoice += price_subtotal * analytic_contribution
-                    else:  # moves_read['move_type'] == 'out_refund'
-                        amount_to_invoice -= price_subtotal * analytic_contribution
-                else:  # moves_read['parent_state'] == 'posted'
-                    if moves_read['move_type'] == 'out_invoice':
-                        amount_invoiced += price_subtotal * analytic_contribution
-                    else:  # moves_read['move_type'] == 'out_refund'
-                        amount_invoiced -= price_subtotal * analytic_contribution
-            # don't display the section if the final values are both 0 (invoice -> credit note)
-            if amount_invoiced != 0 or amount_to_invoice != 0:
-                section_id = 'other_invoice_revenues'
-                invoices_revenues = {
-                    'id': section_id,
-                    'sequence': self._get_profitability_sequence_per_invoice_type()[section_id],
-                    'invoiced': amount_invoiced,
-                    'to_invoice': amount_to_invoice,
-                }
-                if with_action and self.user_has_groups('sales_team.group_sale_salesman_all_leads, account.group_account_invoice, account.group_account_readonly'):
-                    invoices_revenues['action'] = self._get_action_for_profitability_section(list(move_ids), section_id)
-                return {
-                    'data': [invoices_revenues],
-                    'total': {
-                        'invoiced': amount_invoiced,
-                        'to_invoice': amount_to_invoice,
-                    },
-                }
-        return {'data': [], 'total': {'invoiced': 0.0, 'to_invoice': 0.0}}
+            revenues_lines = cogs_lines = []
+            for move_line in invoices_move_line_read:
+                cogs_lines.append(move_line) if move_line['display_type'] == 'cogs' else revenues_lines.append(move_line)
```

Comment:
I think it is more readable without ternary.
```suggestion
                if move_line['display_type'] == 'cogs':
                    cogs_lines.append(move_line)
                else: 
                    revenues_lines.append(move_line)
```